src/safeposix/syscalls/net_calls.rs

```diff
@@ -1234,4 +1233,77 @@ impl Cage {
             return syscall_error(Errno::EBADF, "getsockopt", "the provided file descriptor is invalid");
         }
     }
+
+    pub fn getpeername_syscall(self, fd: i32, ret_addr: &mut interface::GenSockaddr) -> i32 {
+        let fdtable = self.filedescriptortable.read().unwrap();
+
+        if let Some(wrappedfd) = fdtable.get(&fd) {
+            let filedesc = wrappedfd.read().unwrap();
+            if let Socket(socketobj) = &*filedesc {
+                //if the socket is not connected, then we should return an error
+                if socketobj.remoteaddr == None {
+                    return syscall_error(Errno::ENOTCONN, "getpeername", "the socket is not connected");
+                }
+                
+                //all of the checks that we had passed if we are here
+                *ret_addr = socketobj.remoteaddr.unwrap();
+                return 0;
+
+            } else {
+                return syscall_error(Errno::ENOTSOCK, "getpeername", "the provided file is not a socket");
+            }
+        } else {
+            return syscall_error(Errno::EBADF, "getpeername", "the provided file descriptor is not valid");
+        }
+    }
+
+    pub fn getsockname_syscall(self, fd: i32, ret_addr: &mut interface::GenSockaddr) -> i32 {
+        let fdtable = self.filedescriptortable.read().unwrap();
+
+        if let Some(wrappedfd) = fdtable.get(&fd) {
+            let filedesc = wrappedfd.read().unwrap();
+            if let Socket(socketobj) = &*filedesc {
+                if socketobj.localaddr == None {
+                    
+                    //sets the address to 0.0.0.0 if the address is not initialized yet
+                    match ret_addr {
+                        interface::GenSockaddr::V4(_) => {
+                            let addr = interface::GenIpaddr::V4(interface::V4Addr::default());
+                            ret_addr.set_port(0);
+                            ret_addr.set_addr(addr);
+                        }
+        
+                        interface::GenSockaddr::V6(_) => {
+                            let addr = interface::GenIpaddr::V6(interface::V6Addr::default());
+                            ret_addr.set_port(0);
+                            ret_addr.set_addr(addr);
+                        }
```

Comment:
The family is not set for either of these. Also I don't think we can base our assumptions fairly on what was in the ret_addr before. In order to decide what domain it is, use the information in the socketfdobj.domain field. Feel free to add a set_family method to the GenSockaddr object that sets sin_family and sin6_family.

Reply:
Should be done now if I understood the comment correctly.